modules/sfp_countryname.py

```diff
@@ -0,0 +1,323 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_countryname
+# Purpose:      SpiderFoot plug-in for scanning retreived content by other
+#               modules (such as sfp_iban, sfp_phone, sfp_whois) and identifying country names
+#
+# Author:      Krishnasis Mandal <krishnasis@hotmail.com>
+#
+# Created:     28/04/2020
+# Copyright:   (c) Steve Micallef
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import re
+import json
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+import phonenumbers
+from phonenumbers.phonenumberutil import region_code_for_country_code
+
+class sfp_countryname(SpiderFootPlugin):
+    """Country Name Extractor:Footprint,Investigate,Passive:Content Analysis::Identify country names in any obtained data."""
+
+    # Default options
+    opts = {
+        # options specific to this module
+        'coHosted' : True,
+        'affiliate' : True,
+        'nonCountryTLD' : True,
+        'nonCountryTLDDefault' : "United States"
+    }
+
+    # Option descriptions
+    optdescs = {
+        'coHosted' : "Include searching of country name from CO_HOSTED_* data sources",
+        'affiliate' : "Include searching of country name from AFFILIATE_* data sources",
+        'nonCountryTLD' : "Parse TLDs not associated with any country as default country domains",
+        'nonCountryTLDDefault' : "Default value for TLDs not associated with any country"
```

Comment:
The more I think about this, the more I think it doesn't make sense to do for anything beyond .com, .net, etc. since non-country specific domains covers an increasingly broad set of sites such as the new generation of domains (.app, .web, .io, etc.)